manifests/ssltools/nssdb.pp

```diff
@@ -1,50 +1,30 @@
 # Sets up nssdb
 class certs::ssltools::nssdb (
-  $nss_db_dir = $certs::nss_db_dir,
+  $pki_dir = $certs::pki_dir,
+  $nss_db_password = extlib::cache_data('foreman_cache_data', 'certs-nss-db-password', extlib::random_password(32)),
   $group = 'qpidd',
 )  {
   Exec { logoutput => 'on_failure' }
 
-  $nss_db_password_file   = "${nss_db_dir}/nss_db_password-file"
+  $nss_db_dir = "${pki_dir}/nssdb"
+  $nss_db_password_file = "${pki_dir}/nss_db_password-file"
 
-  $nssdb_files = $facts['os']['release']['major'] ? {
-    '7' => ["${nss_db_dir}/cert8.db", "${nss_db_dir}/key3.db", "${nss_db_dir}/secmod.db"],
-    default => ["${nss_db_dir}/cert9.db", "${nss_db_dir}/key4.db", "${nss_db_dir}/pkcs11.txt"]
-  }
-
-  ensure_packages(['openssl', 'nss-tools'])
+  ensure_packages(['nss-tools'])
 
-  file { $nss_db_dir:
-    ensure => directory,
-    owner  => 'root',
-    group  => $group,
-    mode   => '0755',
-  } ->
-  exec { 'generate-nss-password':
-    command => "openssl rand -base64 24 > ${nss_db_password_file}",
-    path    => '/usr/bin',
-    umask   => '0027',
-    group   => $group,
-    creates => $nss_db_password_file,
-    require => Package['openssl'],
-  } ->
   file { $nss_db_password_file:
```

Comment:
You should clean up the old password file. Having unused password files around is confusing:
```suggestion
  file { "${nss_db_dir}/nss_db_password-file":
    ensure => absent,
  }

  file { $nss_db_password_file:
```

Reply:
The old file gets cleaned up automatically, since the password changes, the nssdb is re-created and since it manages the directory it purges any files in there.

Reply:
It's subtle, but: https://github.com/theforeman/puppet-certs/pull/344/commits/a872939829a786b44db6aa1723ed25f8b5ed5060

Since the old password file exists in the NSS DB directory, it will get cleaned up when the NSS DB gets re-created.

Reply:
Technically the new password file does get created before the `nssdb` resource is evaluated (due to the auto-requires).

But the change to extlib cache/generate means the password itself should also change, and it is for this reason that the existence check will always return false the first time, as its performed with a new password.

There is no edge case where a user had previously overridden the password parameter, because it wasn't previously available as a parameter.